server/py/framework/utils/clients/messaging.py

```diff
@@ -312,3 +476,10 @@ def _prepare_request_data(self, method: str, path: str):
         path = path.removeprefix(f"{prefix}/").removeprefix(f"{version}/")
         service_instance = self._discovery.resolve_service_by_request(method, path)
         return path, version, service_instance
+
+    @staticmethod
+    def _resolve_full_request_path(path, service_instance, version):
+        # The service and version prefixes have been removed from the path earlier in the process.
+        # The service prefix will be replaced with the new service name, and the version will be re-added
+        # (or default to v1 if not present) during the final URL construction for the request.
+        return f"{service_instance.url}/{service_instance.name}/{version}/{path}"
```

Comment:
how about using rstrip as we do here? https://github.com/mlrun/mlrun/blob/1b288a97709161603f2586ada547ac1db346ede4/mlrun/common/schemas/api_gateway.py#L106-L109